Makefile.erb

```diff
@@ -0,0 +1,196 @@
+endif
+# }
+
+# Task: composer | alias to run composer within the application container.
+# {
+ifeq (composer,$(firstword $(MAKECMDGOALS)))
+  RUN_ARGS := $(wordlist 2,$(words $(MAKECMDGOALS)),$(MAKECMDGOALS))
+  $(eval $(RUN_ARGS):;@:)
+endif
+
+.PHONY: composer
+composer:
+	@$(DOCKER) exec --user build web composer $(RUN_ARGS); exit 0
+# }
+
+# Task: start | start the application and associated containers.
+# {
+.PHONY: start
+start:
+ifeq ($(OS),macos)
+	@docker ps | grep <%= config.name %>-sync 2>&1 > /dev/null || $(DOCKER_SYNC) start
+endif
+	$(DOCKER) up -d --no-build
+# }
+
+
+# Task: stop | stop the application and associated containers.
+# {
+.PHONY: stop
+stop:
+ifeq ($(OS),macos)
+	$(DOCKER_SYNC) stop || true
+endif
+	$(DOCKER) stop
+# }
+
+# Task: restart | stop and start the application and associated containers.
+# {
+.PHONY: restart
+restart: stop start
+# }
+
+
+# Task: destroy | remove application containers, network and volumes.
+# {
+.PHONY: destroy
+destroy: stop
+	$(DOCKER) down --volumes
+ifeq ($(OS),macos)
+	$(DOCKER_SYNC) clean
+endif
+# }
+
+
+# Task: reset | alias to destroy, build and start the application.
+# {
+.PHONY: reset
+reset: destroy build start
+# }
+
+
+# Task: ip | retrieve IP of web container
+# {
+.PHONY: ip
+ip:
+ifeq ($(OS),linux)
+	@docker inspect -f '{{range .NetworkSettings.Networks}}{{.IPAddress}}{{end}}' $$($(DOCKER) ps -q web)
+endif
+ifneq ($(OS),linux) # Assume docker for mac|windows otherwise
+	@echo 127.0.0.1
+endif
+# }
+
+
+# Task: test | run tests
+# {
+.PHONY: test
+test: test-phpspec
+# }
+
+# Task: test-phpspec | run phpspec examples
+# {
+.PHONY: test-phpspec
+test-phpspec:
+	@echo -e ${INFO} 'running phpspec examples...'
+	@$(DOCKER) exec --user build web bin/phpspec run
+# }
+
+# Task: db-console | open mysql console in the database container
+# {
+.PHONY: db-console
+db-console:
+	@echo -e ${INFO} 'opening mysql console...'
+	@$(DOCKER) exec mysql mysql -u<%= config.name %> -p<%= config.name %> <%= config.name %>
+# }
+
+# Task: db-dump | create db dump
+# {
+ifeq (db-dump,$(firstword $(MAKECMDGOALS)))
+  RUN_ARGS := $(wordlist 2,$(words $(MAKECMDGOALS)),$(MAKECMDGOALS))
+  $(eval $(RUN_ARGS):;@:)
+endif
+
+.PHONY: db-dump
+db-dump:
+	@echo -e ${INFO} 'creating db dump...'
+	$(eval DUMPFILE_NAME=$(if $(RUN_ARGS),$(RUN_ARGS),<%= config.name %>db))
+	$(eval DUMPFILE_NAME="$(DUMPFILE_NAME).sql")
+	@$(DOCKER) exec --user build web mysqldump -u<%= config.name %> -p<%= config.name %> -h mysql <%= config.name %> > $(DUMPFILE_NAME)
+	@echo -e ${INFO} 'db dump saved as' $(DUMPFILE_NAME)
+# }
+
+# Task: db-restore | restores the database to it's initial development state
+# {
+.PHONY: db-restore
+db-restore:
+	@echo -e ${WARNING} 'you are about to restore the database, you will lose data...'; \
+	sleep 3; \
+	$(DOCKER) exec web bash -c 'FORCE_DATABASE_DROP=true container assets_all'
+	@echo -e ${INFO} 'done'
+# }
```

Comment:
/me cries, back to the dark age again of copy and pasting tasks between projects

Reply:
would it be acceptable to continue to write hem plugins instead?

Reply:
(given even docker-sync uses ruby)

Reply:
:+1: will see if we can get this extracted

Reply:
We've extracted this to multiple gems, imported via a Gemfile and a Rakefile.

We are still providing a Makefile to avoid having to say `bundle exec rake <x>` for each of the commands.